CMakeLists.txt

```diff
@@ -66,6 +69,26 @@ else()
   message(STATUS "Will not build OpenJPEG support")
 endif()
 
+set(lib_name ${PROJECT_NAME})
+
+add_library(${lib_name} src/grib2_int.h src/cmplxpack.c src/compack.c
+src/comunpack.c src/drstemplates.c src/g2_addfield.c src/g2_addgrid.c
+src/g2_addlocal.c src/g2_create.c src/g2_free.c src/g2_getfld.c
+src/g2_gribend.c src/g2_info.c src/g2_miss.c src/g2_unpack1.c
+src/g2_unpack2.c src/g2_unpack3.c src/g2_unpack4.c src/g2_unpack5.c
+src/g2_unpack6.c src/g2_unpack7.c src/gbits.c src/getdim.c
+src/getpoly.c src/gridtemplates.c src/int_power.c src/misspack.c
+src/mkieee.c src/pack_gp.c src/pdstemplates.c src/rdieee.c
+src/reduce.c src/seekgb.c src/simpack.c src/simunpack.c src/specpack.c
+src/specunpack.c src/util.c src/g2cxml.c)
+
+set_property(TARGET ${lib_name} PROPERTY C_STANDARD 11)
+
+# Find libxml2.
+find_package(LibXml2 2.9.0 REQUIRED)
+target_link_libraries(${lib_name} PRIVATE ${LIBXML2_LIBRARIES})
+target_include_directories(${lib_name} PRIVATE "${LIBXML2_INCLUDE_DIR}")
```

Comment:
This adds non-optional dependency which makes embedding this project as a submodule in another project that, for example, just needs to decode simple grib messages more difficult. Was this really necessary? We should try to minimize adding more and more dependencies on external libraries.

Reply:
Seems like libxml2 comes installed by default on all our CI systems, so is this really a problem? It will not be a problem at NOAA because libxml2 is already part of our installed stack.

The netcdf-c library recently added a dependency on libxml2, with no reported problems from users.

It's not clear to me whether this will be an optional feature in the next release. I am still exploring how this will be used.

Reply:
Availability of a library shouldn't be excuse for adding it as a dependency. Also the fact that netcdf-c depends on it, that's irrelevant in this case.